syncd/syncd.cpp

```diff
@@ -2954,25 +2957,98 @@ void processFlexCounterGroupEvent(
     }
 }
 
+std::queue<swss::KeyOpFieldsValuesTuple> g_flexCounterEventQueue;
+
+bool tryPopFlexCounterEvent(
+        _Out_ swss::KeyOpFieldsValuesTuple& kco)
+{
+    SWSS_LOG_ENTER();
+
+    std::lock_guard<std::mutex> lock(g_mutex);
+
+    if (g_flexCounterEventQueue.empty())
+    {
+        return false;
+    }
+
+    kco = g_flexCounterEventQueue.front();
+
+    g_flexCounterEventQueue.pop();
+
+    return true;
+}
+
+void pushFlexCounterEvent(
+    _In_ const swss::KeyOpFieldsValuesTuple& kco)
+{
+    SWSS_LOG_ENTER();
+
+    std::lock_guard<std::mutex> lock(g_mutex);
+
+    g_flexCounterEventQueue.push(kco);
+}
+
+bool processFlexCounterEvent(
+    _In_ const swss::KeyOpFieldsValuesTuple kco);
+
+void processFlexCounterEventThread()
+{
+    SWSS_LOG_ENTER();
+
+    while (g_processFlexCounterEventThreadRun)
+    {
+        swss::KeyOpFieldsValuesTuple kco;
+
+        if (tryPopFlexCounterEvent(kco))
+        {
+            if (!processFlexCounterEvent(kco))
+            {
+                // event was not successfully processed, put it again to the queue
+
+                pushFlexCounterEvent(kco);
+            }
+        }
+
+        sleep(1);
```

Comment:
Can we wake up every 1 second only if there is a processing event failure?  In other cases, we wake up on an enqueue event. Flex counter table event should be very rare after the initialization.

Reply:
this is actually overkill to make other thread and condition variable to wait for specific event, also making that event, also don't guarantee that this event will be processed after second select will create rif port, so there must be wait somewhere anyway